### PR TITLE
WIP: Record Binary Search Refactoring

### PR DESCRIPTION
**Pull request by prathje**
This PR should eliminate a few potential issues with the current binary search.
1. Correct handling of sn_increment_by (as it only supports unsigned ints)
2. Add method to get a reliable start and end sequence numbers using storage's lock mechanism
3. Guarantee that the min and max (e.g. for start and end) will really result in the first and last of the entries with that timestamps. 

Todos:
- [ ] Add tests (by sequence / by timerange, in case of empty records, full records, missing records etc.)

### CODE REVIEW

**Comment by prathje** [comment]
Blocked by #16 for testing.

File: src/ens/sequencenumber.c [hunk 1]
```diff
@@ -7,18 +7,30 @@
...
+    } else {
+        return SN_MASK;
+    }
+}
+
 record_sequence_number_t sn_increment_by(record_sequence_number_t sn, uint32_t amount) {
     return GET_MASKED_SN((sn + amount));
 }
 
+record_sequence_number_t sn_decrement_by(record_sequence_number_t sn, uint32_t amount) {
```
Comment (H1ghBre4k3r) [inline src/ens/sequencenumber.c]: I think, we should add a corresponding function declaration in `sequencenumber.h`.

Comment (prathje) [reply src/ens/sequencenumber.c]: Oh, this function is actually not used anymore ;)
I'll remove it!

**Review by H1ghBre4k3r — APPROVED** [review]
